File: solutions/zig/armstrong-numbers/1/armstrong_numbers.zig
const std = @import("std");

pub fn isArmstrongNumber(num: u128) bool {
    var n = num;
    var sum: u128 = 0;
    var len: u128 = 0;

    while (n > 0) : (n /= 10) {
        len += 1;
    }
    n = num;

    while (n > 0) : (n /= 10) {
        sum += std.math.pow(u128, n % 10, len);
    }
    return sum == num;
}

File: solutions/go/armstrong-numbers/1/armstrong_numbers.go
package armstrong

import "math"

func IsNumber(n int) bool {
	num := n
	sum := 0
	len := 0

	for num > 0 {
		len++
		num /= 10
	}
	num = n

	for num > 0 {
		sum += int(math.Pow(float64(num%10), float64(len)))
		num /= 10
	}
	return sum == n
}
